### PR TITLE
Rework paradrop function to work better

### PR DESCRIPTION
**Pull request by wurzel0701**
## What type of PR is this.
1. [X] Bug
2. [X] Change
3. [X] Enhancement

### What have you changed and why?
Information:
Changed the functionality of the paratroopers from getting shot while dangling in the sky to actual fast insertions by falling until they nearly reach the ground, then pulling the parachute shortly above the ground.

Also created the header, fixed a bug that made attacks useless on current unstable and made some performance tweaks.

Execution can be seen here 
https://youtu.be/5tIpo9IOUsA?t=63

Keep in mind that AI has no chance against statics, so they never had a chance as the air supports is nearly useless.
    
### Please specify which Issue this PR Resolves.
None opened

### Please verify the following and ensure all checks are completed.

1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: Calling in an attack (airfield to airfield works best as it ensures paradrops) and see how it works

********************************************************
Notes: Not tested on dedicated server but should be working there. Increases difficulty for players heavily as these paratroopers are hard to see at day and nearly invisible in the night, might needs balancing.

### CODE REVIEW

File: A3-Antistasi/functions/AI/fn_paradrop.sqf [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+private _exitPos = _targetPosition getPos [_entryDistance, _normalAngle + _attackAngle];
+
+{
+    _x set [2,300];
+} forEach [_entryPos, _exitPos, _originPosition];
+
+private _wp = _groupPilot addWaypoint [_entryPos, -1];
+_wp setWaypointType "MOVE";
+_wp setWaypointSpeed "NORMAL";
+_wp setWaypointStatements ["true", "if !(local this) exitWith {}; (vehicle this) setVariable ['dropPosReached', true]; [3, 'Drop pos reached', 'paradrop'] call A3A_fnc_log;"];
```
Comment (HakonRydland) [inline A3-Antistasi/functions/AI/fn_paradrop.sqf:75]: recommend use of new macros, also can move the log to inside the while loop after ``if(_vehicle getVariable ["dropPosReached", false]) then``

File: A3-Antistasi/functions/AI/fn_paradrop.sqf [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+private _groupPilot = group driver _vehicle;
+{
+    _x disableAI "TARGET";
+    _x disableAI "AUTOTARGET";
+    _x setBehaviour "CARELESS";
+} foreach (units _groupPilot);
+
+{
+    removebackpack _x;
+} forEach (units _groupJumper);
```
Comment (HakonRydland) [inline A3-Antistasi/functions/AI/fn_paradrop.sqf:43]: consider storing backpack and contents for restoration after drop.

**Review by HakonRydland — CHANGES_REQUESTED** [review]
2 changes requested, one recommendation, one to consider

File: A3-Antistasi/functions/AI/fn_paradrop.sqf [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+{
+    _vehicle setCollisionLight true;
+	{
+        unAssignVehicle _x;
+        //Move them into alternating left/right positions, so their parachutes are less likely to kill each other
+        private _pos = if (_forEachIndex % 2 == 0) then {_vehicle modeltoWorld [7, -20, -5]} else {_vehicle modeltoWorld [-7, -20, -5]};
+        _x setPos _pos;
+        _x spawn
+        {
+            waitUntil {sleep 0.25; ((getPosATL _this) select 2) < 105};
```
[the remark this answers is not in the document]
Comment (wurzel0701) [reply A3-Antistasi/functions/AI/fn_paradrop.sqf]: Fixed

File: A3-Antistasi/functions/AI/fn_paradrop.sqf (hunk 2 above)
Comment (wurzel0701) [reply A3-Antistasi/functions/AI/fn_paradrop.sqf]: Added, but that was not part of the original functionality (they seem to be more effective in combat tho)

File: A3-Antistasi/functions/AI/fn_paradrop.sqf [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            _this addBackpack "B_Parachute";
+            private _smokeGrenade = selectRandom allSmokeGrenades;
+    		private _smoke = _smokeGrenade createVehicle (getPosATL _this);
```
[the remark this answers is not in the document]
Comment (wurzel0701) [reply A3-Antistasi/functions/AI/fn_paradrop.sqf]: Fixed

**Review by HakonRydland — APPROVED** [review]
looks good

**Review by Lazejun — APPROVED** [review]
From a Gameplay perspective, i can confirm Paradrops work pretty well,
Backpacks get saved as intended by the Code,
Helicopters and Transportplanes seem to have a higher chance of survival,
And most importantly we don't have the issue of Helicopters lacking Waypoints.

**Pull request merged**